resource/resource.go

```diff
@@ -57,6 +57,11 @@ type Extra struct {
 	Resource *unstructured.Unstructured
 }
 
+// Credential is a secret requested by a Function
+type Credential struct {
+	Data *unstructured.Unstructured
```

Comment:
This can just be a `map[string][]byte` rather than `*Unstructured`. `*Unstructured` is intended to model a Kubernetes object, but Data won't be one - it's going to be a `map[string][]byte`.

Reply:
Indeed make sense